ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/parser/SerialTreeParser.java

```diff
@@ -130,4 +134,19 @@ private boolean isModuleDeclarationAllowed(ModuleMemberDeclarationNode declarati
         }
         return true;
     }
+
+    /**
+     * Whether source is allowed to be parsed
+     */
+    private boolean isSourceAllowed(String source) {
+        Optional<String> matchedSource = ParserConstants.getMatchedRestrictedSource(source);
+
+        if (matchedSource.isPresent()) {
+            addWarnDiagnostic("Found restricted keyword '" + matchedSource.get() + "' in the declaration.\n" +
+                    "Discarded '" + source + "' without loading.");
+            return false;
+        }
+
```

Comment:
I would prefer to add new lines after a if/for condition with huge length

Reply:
Thanks for the suggestion Sasindu. I removed the newline.